src/Summoner/License.hs

```diff
@@ -68,13 +68,14 @@ githubLicenseQueryNames = \case
 parseLicenseName :: Text -> Maybe LicenseName
 parseLicenseName = inverseMap show
 
-customizeLicense :: LicenseName -> Text -> Text -> Text -> Text
+customizeLicense :: LicenseName -> License -> Text -> Text -> License
 customizeLicense l t nm year
```

Comment:
instead of `t` you can write `licence@(License licenseText)`, so no need to add another function in `where` clause